### PR TITLE
Implement fmt.sh --diff for the Java style check

### PR DESCRIPTION
**Pull request by stefanobaghino-da**
When run with --diff, fmt.sh now only checks for the .java files that have been
changed since the fork point. This makes local git pre-push hooks more quiet
while preserving the safe CI behavior (which doesn't use --diff).

changelog_begin
changelog_end

### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [ ] Include appropriate tests
- [x] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [x] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

### CODE REVIEW

**Comment by stefanobaghino-da** [comment]
I tested that the changes behaved as expected on Java files using various combinations of with and without `--test` and `--diff`.

What is an `hlint` violation that I can use to make sure I'm not breaking anything there?

**Comment by cocreature** [comment]
> What is an `hlint` violation that I can use to make sure I'm not breaking anything there?

redundant parentheses iirc

File: fmt.sh [hunk 1]
```diff
@@ -112,24 +125,21 @@ run dade-copyright-headers "$dade_copyright_arg" .
...
-  fi
+if [ "$diff_mode" = "true" ]; then
+  check_diff $merge_base '\.hs$' hlint -j4
+  check_diff $merge_base '\.java$' javafmt "${javafmt_args[@]:-}"
 else
-  "${HLINT[@]}" --git
+  run hlint -j4 --git
+  java_files=$(find . -name "*.java")
+  if [[ -z "$java_files" ]]; then
+    echo "Unexpected: no Java file in the repository"
```
Comment (cocreature) [inline fmt.sh:135]: accepting bets for when we’ll reach that.

**Review by cocreature — APPROVED** [review]
lgtm, thanks!

File: fmt.sh (hunk 1 above)
Comment (stefanobaghino-da) [reply fmt.sh:135]: Do you have a Daml driven betting platform? 😉

**Comment by stefanobaghino-da** [comment]
> > What is an `hlint` violation that I can use to make sure I'm not breaking anything there?
> 
> redundant parentheses iirc

Tested, works. 🙇🏻

**Pull request merged**